signaler/signaler_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build !windows
```

Comment:
🚧 This feels wrong.
I don't think we should be avoiding building a test.
I think we should skip the tests inside it based on the platform instead.
If the entire feature was optional, we'd use build tags, but this doesn't feel right.